### PR TITLE
Translation zh

### PR DESCRIPTION
**Pull request by xiaod-dev**
Pull request description

- add language code 
- correct pic path

### CODE REVIEW

**Comment by xiaod-dev** [comment]
@iluwatar 
I found a translation tool https://crowdin.com/, It seems more industrial and provide more Translate efficiently.
It not free, But It shows free of charge for open-source projects.
https://crowdin.com/page/open-source-project-setup-request
The [ethereum ](https://ethereum.org/en/)project  is also using this for translation work.
I think we can think about it.

File: localization/zh/builder/README.md [hunk 1]
```diff
@@ -114,7 +114,7 @@ var mage = new Hero.Builder(Profession.MAGE, "Riobard").withHairColor(HairColor.
```
Comment (ohbus) [inline localization/zh/builder/README.md:114]: Please fix the **permalink** for this file

File: localization/zh/caching/README.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ tags:
 为了避免昂贵的资源重新获取，方法是在资源使用后不立即释放资源。资源保留其身份，保留在某些快速访问的存储中，并被重新使用，以避免再次获取它们。
 
 ## 类图
-![alt text](../../caching/etc/caching.png "Caching")
+![alt text](../../../caching/etc/caching.png "Caching")
```
Comment (ohbus) [inline localization/zh/caching/README.md:17]: Please fix the **permalink** for this file

File: localization/zh/callback/README.md [hunk 3]
```diff
@@ -69,7 +69,7 @@ public final class SimpleTask extends Task {
     task.executeWith(() -> LOGGER.info("I'm done now."));
 ```
 ## 类图
-![alt text](../../callback/etc/callback.png "Callback")
+![alt text](../../../callback/etc/callback.png "Callback")
```
Comment (ohbus) [inline localization/zh/callback/README.md:72]: Please fix the **permalink** for this file

File: localization/zh/chain/README.md [hunk 4]
```diff
@@ -139,7 +139,7 @@ king.makeRequest(new Request(RequestType.COLLECT_TAX, "collect tax")); // Orc so
 ```
 
 ## 类图
-![alt text](../../chain/etc/chain.urm.png "Chain of Responsibility class diagram")
+![alt text](../../../chain/etc/chain.urm.png "Chain of Responsibility class diagram")
```
Comment (ohbus) [inline localization/zh/chain/README.md:142]: Please fix the **permalink** for this file

File: localization/zh/command/README.md [hunk 5]
```diff
@@ -222,7 +222,7 @@ goblin.printStatus();
 ```
 
 ## 类图
-![alt text](../../command/etc/command.png "Command")
+![alt text](../../../command/etc/command.png "Command")
```
Comment (ohbus) [inline localization/zh/command/README.md:225]: Please fix the **permalink** for this file

File: localization/zh/composite/README.md [hunk 6]
```diff
@@ -154,7 +154,7 @@ elfMessage.print(); // Much wind pours from your mouth.
 
 ## 类图
 
-![alt text](../../composite/etc/composite.urm.png "Composite class diagram")
+![alt text](../../../composite/etc/composite.urm.png "Composite class diagram")
```
Comment (ohbus) [inline localization/zh/composite/README.md:157]: Please fix the **permalink** for this file

File: localization/zh/converter/README.md [hunk 7]
```diff
@@ -86,7 +86,7 @@ var user = userConverter.convertFromDto(dtoUser);
 
 ## 类图
 
-![alt text](../../converter/etc/converter.png "Converter Pattern")
+![alt text](../../../converter/etc/converter.png "Converter Pattern")
```
Comment (ohbus) [inline localization/zh/converter/README.md:89]: Please fix the **permalink** for this file

File: localization/zh/dao/README.md [hunk 8]
```diff
@@ -149,7 +149,7 @@ customerDao.getAllCustomers(): java.util.stream.ReferencePipeline$Head@6ec8211c
 
 ## 类图
 
-![alt text](../../dao/etc/dao.png "Data Access Object")
+![alt text](../../../dao/etc/dao.png "Data Access Object")
```
Comment (ohbus) [inline localization/zh/dao/README.md:152]: Please fix the **permalink** for this file

File: localization/zh/facade/README.md [hunk 9]
```diff
@@ -189,7 +189,7 @@ facade.endDay();
 ```
 
 ## 类图
-![alt text](../../facade/etc/facade.urm.png "Facade pattern class diagram")
+![alt text](../../../facade/etc/facade.urm.png "Facade pattern class diagram")
```
Comment (ohbus) [inline localization/zh/facade/README.md:192]: Please fix the **permalink** for this file

File: localization/zh/dirty-flag/README.md [hunk 10]
```diff
@@ -17,7 +17,7 @@ tags:
 避免昂贵资源的重新获取。资源保留其身份，保留在某些快速访问的存储中，并被重新使用以避免再次获取它们。
 
 ## 类图
-![alt text](../../dirty-flag/etc/dirty-flag.png "Dirty Flag")
+![alt text](../../../dirty-flag/etc/dirty-flag.png "Dirty Flag")
```
Comment (ohbus) [inline localization/zh/dirty-flag/README.md:20]: Please fix the **permalink** for this file

File: localization/zh/dependency-injection/README.md [hunk 11]
```diff
@@ -84,7 +84,7 @@ public class AdvancedWizard implements Wizard {
 
 ## 类图
 
-![alt text](../../dependency-injection/etc/dependency-injection.png "Dependency Injection")
+![alt text](../../../dependency-injection/etc/dependency-injection.png "Dependency Injection")
```
Comment (ohbus) [inline localization/zh/dependency-injection/README.md:87]: Please fix the **permalink** for this file

File: localization/zh/delegation/README.md [hunk 12]
```diff
@@ -16,7 +16,7 @@ tags:
 它是一种让对象将某种行为向外部表达，但实际上将实现该行为的责任委托给关联对象的技术。
 
 ## 类图
-![alt text](../../delegation/etc/delegation.png "Delegate")
+![alt text](../../../delegation/etc/delegation.png "Delegate")
```
Comment (ohbus) [inline localization/zh/delegation/README.md:19]: Please fix the **permalink** for this file

File: localization/zh/decorator/README.md [hunk 13]
```diff
@@ -106,7 +106,7 @@ clubbedTroll.fleeBattle(); // The troll shrieks in horror and runs away!
 ```
 
 ## 类图
-![alt text](../../decorator/etc/decorator.urm.png "Decorator pattern class diagram")
+![alt text](../../../decorator/etc/decorator.urm.png "Decorator pattern class diagram")
```
Comment (ohbus) [inline localization/zh/decorator/README.md:109]: Please fix the **permalink** for this file

File: localization/zh/data-transfer-object/README.md [hunk 14]
```diff
@@ -94,7 +94,7 @@ public class CustomerResource {
 
 ## 类图
 
-![alt text](../../data-transfer-object/etc/data-transfer-object.urm.png "data-transfer-object")
+![alt text](../../../data-transfer-object/etc/data-transfer-object.urm.png "data-transfer-object")
```
Comment (ohbus) [inline localization/zh/data-transfer-object/README.md:97]: Please fix the **permalink** for this file

File: localization/zh/factory-method/README.md [hunk 15]
```diff
@@ -63,7 +63,7 @@ blacksmith.manufactureWeapon(WeaponType.AXE);
 ```
 
 ## 类图
-![alt text](../../factory-method/etc/factory-method.urm.png "Factory Method pattern class diagram")
+![alt text](../../../factory-method/etc/factory-method.urm.png "Factory Method pattern class diagram")
```
Comment (ohbus) [inline localization/zh/factory-method/README.md:66]: Please fix the **permalink** for this file

File: localization/zh/proxy/README.md [hunk 16]
```diff
@@ -123,7 +123,7 @@ Brown wizard is not allowed to enter!
 
 ## 类图
 
-![alt text](../../proxy/etc/proxy.urm.png "Proxy pattern class diagram")
+![alt text](../../../proxy/etc/proxy.urm.png "Proxy pattern class diagram")
```
Comment (ohbus) [inline localization/zh/proxy/README.md:126]: Please fix the **permalink** for this file

File: localization/zh/producer-consumer/README.md [hunk 17]
```diff
@@ -13,7 +13,7 @@ tags:
 生产者消费者设计模式是一种经典的并发模式，通过将工作与执行工作任务分开来减少生产者与消费者之间的耦合。
 
 ## 类图
-![alt text](../../producer-consumer/etc/producer-consumer.png "Producer Consumer")
+![alt text](../../../producer-consumer/etc/producer-consumer.png "Producer Consumer")
```
Comment (ohbus) [inline localization/zh/producer-consumer/README.md:16]: Please fix the **permalink** for this file

File: localization/zh/private-class-data/README.md [hunk 18]
```diff
@@ -120,7 +120,7 @@ immutableStew.mix();  // Mixing the immutable stew we find: 2 potatoes, 4 carrot
 
 ## 类图
 
-![alt text](../../private-class-data/etc/private-class-data.png "Private Class Data")
+![alt text](../../../private-class-data/etc/private-class-data.png "Private Class Data")
```
Comment (ohbus) [inline localization/zh/private-class-data/README.md:123]: Please fix the **permalink** for this file

File: localization/zh/observer/README.md [hunk 19]
```diff
@@ -129,7 +129,7 @@ public class Weather {
 ```
 
 ## Class diagram
-![alt text](../../observer/etc/observer.png "Observer")
+![alt text](../../../observer/etc/observer.png "Observer")
```
Comment (ohbus) [inline localization/zh/observer/README.md:132]: Please fix the **permalink** for this file

File: localization/zh/iterator/README.md [hunk 20]
```diff
@@ -115,7 +115,7 @@ while (itemIterator.hasNext()) {
 ```
 
 ## 类图
-![alt text](../../iterator/etc/iterator_1.png "Iterator")
+![alt text](../../../iterator/etc/iterator_1.png "Iterator")
```
Comment (ohbus) [inline localization/zh/iterator/README.md:118]: Please fix the **permalink** for this file

File: localization/zh/interpreter/README.md [hunk 21]
```diff
@@ -13,7 +13,7 @@ tags:
 给定一种语言，请定义其语法的表示形式，以及使用该表示形式来解释该语言中的句子的解释器。
 
 ## 类图
-![alt text](../../interpreter/etc/interpreter_1.png "Interpreter")
+![alt text](../../../interpreter/etc/interpreter_1.png "Interpreter")
```
Comment (ohbus) [inline localization/zh/interpreter/README.md:16]: Please fix the **permalink** for this file

File: localization/zh/state/README.md [hunk 22]
```diff
@@ -135,7 +135,7 @@ public class Mammoth {
 ```
 
 ## 类图
-![alt text](../../state/etc/state_urm.png "State")
+![alt text](../../../state/etc/state_urm.png "State")
```
Comment (ohbus) [inline localization/zh/state/README.md:138]: Please fix the **permalink** for this file

File: localization/zh/strategy/README.md [hunk 23]
```diff
@@ -112,7 +112,7 @@ public class DragonSlayer {
 ```
 
 ## 类图
-![alt text](../../strategy/etc/strategy_1.png "Strategy")
+![alt text](../../../strategy/etc/strategy_urm.png "Strategy")
```
Comment (ohbus) [inline localization/zh/strategy/README.md:115]: Please fix the **permalink** for this file

File: localization/zh/template-method/README.md [hunk 24]
```diff
@@ -119,7 +119,7 @@ public class HalflingThief {
 ```
 
 ## 类图
-![alt text](../../template-method/etc/template_method_urm.png "Template Method")
+![alt text](../../../template-method/etc/template_method_urm.png "Template Method")
```
Comment (ohbus) [inline localization/zh/template-method/README.md:122]: Please fix the **permalink** for this file

File: localization/zh/visitor/README.md [hunk 25]
```diff
@@ -201,7 +201,7 @@ Good to see you commander
 
 ## Class diagram
 
-![alt text](../../visitor/etc/visitor_1.png "Visitor")
+![alt text](../../../visitor/etc/visitor_1.png "Visitor")
```
Comment (ohbus) [inline localization/zh/visitor/README.md:204]: Please fix the **permalink** for this file

**Review by ohbus — CHANGES_REQUESTED** [review]
As you are aware of the recent changes in localization management.

Please update the **permalink** for the following files by removing the `zh` at the end of it.

**Review by ohbus — APPROVED** [review]
LGTM

**Comment by iluwatar** [comment]
> @iluwatar
> I found a translation tool https://crowdin.com/, It seems more industrial and provide more Translate efficiently.
> It not free, But It shows free of charge for open-source projects.
> https://crowdin.com/page/open-source-project-setup-request
> The [ethereum ](https://ethereum.org/en/)project is also using this for translation work.
> I think we can think about it.

@xiaod-dev thanks for the research effort. I agree we should evaluate the tool. I created a new issue https://github.com/iluwatar/java-design-patterns/issues/1809 about it.

**Pull request merged**